test/search_api/endpoints/test_search_endpoint.py

```diff
@@ -0,0 +1,402 @@
+                                "rangeTop": 101.0,
+                                "error": 31472.0,
+                            },
+                        ],
+                        "name": "INVESTIGATION 1",
+                        "releaseDate": "2000-07-05 00:00:00+00:00",
+                        "id": 1,
+                    },
+                ],
+                id="Search documents including all possible related entities",
+                # Skipped because of incorrect members JSON naming
+                # (investigationUsers key used instead). It's in ICAT fields too,
+                # not converted into PaNOSC format
+                marks=pytest.mark.skip,
+            ),
+            pytest.param(
+                "instruments",
+                '{"include": [{"relation": "datasets"}], "limit": 1}',
+                {
+                    "description": "Former outside source play nearly Congress before"
+                    " necessary. Allow want audience test laugh. Economic body person"
+                    " general attorney. Effort weight prevent possible.",
+                    "modId": "user",
+                    "createTime": "2019-02-19 05:57:03+00:00",
+                    "pid": None,
+                    "createId": "user",
+                    "type": "2",
+                    "name": "INSTRUMENT 2",
+                    "modTime": "2019-01-29 23:33:20+00:00",
+                    "id": 2,
+                    "fullName": "With piece reason late model. House office fly."
+                    " International scene call deep answer audience baby true.\n"
+                    "Indicate education across these. Opportunity design too.",
+                    "url": "https://moore.org/",
+                },
+                id="Search instruments including all possible related entities",
+                # Skipped due to ICAT 5 mapping
+                marks=pytest.mark.skip,
+            ),
+            pytest.param(
+                "datasets",
+                '{"limit": 1, "where": {"isPublic": true}}',
+                [{}],
+                id="Search datasets with isPublic condition",
+                # Skipped because the where for isPublic doesn't work
+                marks=pytest.mark.skip,
+            ),
+        ],
+    )
+    def test_valid_search_endpoint(
+        self, flask_test_app_search_api, endpoint_name, request_filter, expected_json,
+    ):
+        test_response = flask_test_app_search_api.get(
+            f"{Config.config.search_api.extension}/{endpoint_name}?filter="
+            f"{request_filter}",
+        )
+
+        assert test_response.json == expected_json
+
+    @pytest.mark.parametrize(
+        "request_filter, expected_status_code",
+        [
+            pytest.param('{"where": []}', 400, id="Bad where filter"),
+            pytest.param('{"limit": -1}', 400, id="Bad limit filter"),
+            pytest.param('{"limit": -100}', 400, id="Bad skip filter"),
```

Comment:
This should be `skip`
```suggestion
            pytest.param('{"skip": -100}', 400, id="Bad skip filter"),
```

Reply:
Committed all these together, I should've checked it before I did some copy+paste work ;)